x-pack/plugins/observability_solution/observability_onboarding/public/application/onboarding_flow_form/onboarding_flow_form.tsx

```diff
@@ -126,17 +106,16 @@ export const OnboardingFlowForm: FunctionComponent = () => {
   const createCollectionCardHandler = useCallback(
     (query: string) => () => {
       setIntegrationSearch(query);
-      if (packageListRef.current) {
-        // adding a slight delay causes the search bar to be rendered
-        new Promise((r) => setTimeout(r, 10)).then(() =>
-          packageListRef.current?.scrollIntoView({
-            behavior: 'smooth',
-            block: 'start',
-          })
+      if (searchResultsRef.current) {
+        setTimeout(
```

Comment:
It might make sense to put `setTimeout` right into the `scrollIntoViewWithOffset`, they seem like part of the same logic, wdyt?

Reply:
Not necessarily, the only reason we need the timeout is because we're toggling DOM elements. If the elements would already be rendered on the page we could scroll straight to them without a delay so I would leave that logic separate.